core/meta/src/TDataType.cxx

```diff
@@ -422,8 +422,8 @@ void TDataType::AddBuiltins(TCollection* types)
       fgBuiltins[kULong_t] = new TDataType("unsigned long");
       fgBuiltins[kLong64_t] = new TDataType("long long");
       fgBuiltins[kULong64_t] = new TDataType("unsigned long long");
-      fgBuiltins[kFloat_t] = new TDataType("float");
-      fgBuiltins[kDouble_t] = new TDataType("double");
+      fgBuiltins[kFloat_t] = fgBuiltins[kFloat16_t] = new TDataType("float");
+      fgBuiltins[kDouble_t] = fgBuiltins[kDouble32_t] = new TDataType("double");
```

Comment:
`kFloat_t` and `kFloat16_t `(and `Double_t` and `Double32_t`) are *not* interchangeble`.  See other use in the same file.
```suggestion
      fgBuiltins[kFloat_t] = new TDataType("float");
      fgBuiltins[kDouble_t] = new TDataType("double");
      fgBuiltins[kFloat16_t] = new TDataType("Float16_t");
      fgBuiltins[kDouble32_t] = new TDataType("Double32_t");
```
might work.